Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+    if (SecureEntry) {
+        PasswordBox->IsEnabled = false;
+        PasswordBox->IsEnabled = true;
+    } else {
+        TextBox->IsEnabled = false;
+        TextBox->IsEnabled = true;
+    }
+}
+
+void TextField::SetFocus() {
+    if (SecureEntry) {
+        PasswordBox->Focus(FocusState::Programmatic);
+    } else {
+        TextBox->Focus(FocusState::Programmatic);
+    }
+}
+
+bool TextField::BecomeFirstResponder() {
+    if (SecureEntry) {
+        return PasswordBox->Focus(FocusState::Programmatic);
+    } else {
+        return TextBox->Focus(FocusState::Programmatic);
+    }
+}
+
+void TextField::OnTextChanged(Platform::Object ^sender, Windows::UI::Xaml::Controls::TextChangedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnPasswordChanged(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnGotFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_gotFocusHandler) {
+        _gotFocusHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnLostFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_lostFocusHanlder) {
+        _lostFocusHanlder->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnKeyDown(Platform::Object ^sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs ^e)
+{
+    if (_enterKeyDownHandler) {
+        if (e->Key == Windows::System::VirtualKey::Enter) {
+            _enterKeyDownHandler->Invoke(
+                InspectableFromObject(this).Get(),
+                ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+            e->Handled = true;
+        } else {
+            e->Handled = false;
+        }
+    }
+}
+
+void TextField::ApplyFont(Platform::String^ fontFamily, FontStretch strech, FontStyle fontStyle, double fontSize, FontWeight fontWeight) {
+    if (SecureEntry) {
+        PasswordBox->FontFamily = ref new FontFamily(fontFamily);
+        PasswordBox->FontStretch = strech;
+        PasswordBox->FontStyle = fontStyle;
+        PasswordBox->FontSize = fontSize;
+        PasswordBox->FontWeight = fontWeight;
+    }
+    else {
+        TextBox->FontFamily = ref new FontFamily(fontFamily);
+        TextBox->FontStretch = strech;
+        TextBox->FontStyle = fontStyle;
+        TextBox->FontSize = fontSize;
+        TextBox->FontWeight = fontWeight;
+    }
```

Comment:
nice, didn't realize this in on control.

---
In reply to: [113563428](https://github.com/Microsoft/WinObjC/pull/2616#discussion_r113563428) [](ancestors = 113563428)